lib/jekyll/site.rb

```diff
@@ -68,6 +68,7 @@ def reset
       self.static_files = []
       self.data = {}
       @collections = nil
+      @regenerator.clear_cache()
```

Comment:
Ah, yes... can you tell it's been a while since I last coded in ruby? (^_^)
I'll fix this and regenerator clear tonight.